admiral/pkg/controller/admiral/deployment_test.go

```diff
@@ -68,6 +68,7 @@ func TestDeploymentController_Added(t *testing.T) {
 	}
 	for _, c := range testCases {
 		t.Run(c.name, func(t *testing.T) {
+			depController.K8sClient = fake.NewSimpleClientset()
```

Comment:
Can you add test cases for deployment annotation and namespace annotation? you can use fake client to add these combinations.